src/carma-streets/plot_message_frequencies.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+import argparse
+from pathlib import Path
+import sys
+from parse_kafka_logs import KafkaLogMessageType
+
+from matplotlib import pyplot as plt
+from matplotlib import axes
+import pandas as pd
+
+def plot_message_frequencies(csv_dir: Path, plots_dir: Path, simulation: bool = False):
+    """Function to plot all message frequencies in a single figure
+
+    Args:
+        csv_dir (Path): directory that holds CSV message data.
+        plots_dir (Path): directory where create figure will be saved
+        simulation (bool,optional): bool flag to indicate whether data was collected in simulation. Defaults to False.
+    """
+    if csv_dir.is_dir():
+        if plots_dir.is_dir():
+            print(f"WARNING: {plots_dir} already exists. Contents will be overwritten.")
+        plots_dir.mkdir(exist_ok=True)
+        # Read CSV data
+        message_data = dict([])
+        time_sync_data = None
+        for csv_file in csv_dir.glob("*.csv"):
+            print(f'Reading csv file {csv_file.name} ...')
+            if csv_file.name == 'time_sync.csv':
+                time_sync_data = pd.read_csv(csv_file)
+            else:
+                df = pd.read_csv(csv_file)
+                message_data[csv_file.stem] = df
+        fig, plots = plt.subplots(len(message_data), sharex=True, layout="constrained")
+        # Add simulation time to message data
+        for msg_plot,( message_name, message_data_frame) in zip( plots, message_data.items()):
+            print(f'Getting simulation time for {message_name} data ...')
+            if simulation:
+                message_data_frame["Time (ms)"] = get_simulation_time(message_data_frame["Created Time(ms)"], time_sync_data["Created Time(ms)"], time_sync_data["Timestamp(ms)"])
+                message_data_frame["Time (s)"] = message_data_frame["Time (ms)"]/1000
+            else :
+                message_data_frame["Time (s)"] = message_data_frame["Timestamp(ms)"]/1000
+            message_data_frame = add_message_frequency_columns(message_data_frame)
```

Comment:
You do not need to reassign here. The `add_message_frequency_columns()` function should not return anything.

Reply:
Oh python passes by reference by default? Is there a pythonic way to indicate in a comment that we are modifying passed in parameters.

Reply:
I think the Pythonic way would be to generate a new `DataFrame` or `Series` containing the data generated from your immutable messages `DataFrame`s. However, I think the changing the function's name would be sufficient for our use here.